Updates to tgmath functions for CGFloat

stephentyrone opened the pull request:
These changes bring CGFloat in line with the other FloatingPoint types.  Some of this stuff is now defined at the protocol level, so we can get rid of it at the concrete type level, with an obsoleted message.  A couple other functions have been deprecated for all types, with protocol or tgmath replacements.

stephentyrone committed:
Updates to tgmath functions for CGFloat

These changes bring CGFloat in line with the other FloatingPoint types.  Some of this stuff is now defined at the protocol level, so we can get rid of it at the concrete type level, with an obsoleted message.  A couple other functions have been deprecated for all types, with protocol or tgmath replacements.

jrose-apple commented on stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb:
```diff
@@ -690,28 +551,36 @@ public func isnan(_ value: CGFloat) -> Bool { return value.isNaN }
   return (CGFloat(ipart), CGFloat(fpart))
 }
 
+@available(swift, deprecated: 4.2, renamed: "scalbn")
```
It's weird to have something deprecated then provide a rename to something obsoleted in the same version.

stephentyrone replied:
`scalbn` is obsoleted for concrete types; there's a new `scalbn` that's generic on `BinaryFloatingPoint`.

jrose-apple replied:
Ah! Thanks for the explanation.

jrose-apple replied:
Any reason we can't just remove them, then?

stephentyrone replied:
Not that I know of; ben and max recommended `obsoleted`. If you think removal is viable, I'll make another PR for that.

jrose-apple commented on test/stdlib/tgmath.swift.gyb:
```diff
@@ -189,8 +189,8 @@ extension ${T}: TGMath {
-% if T == 'Double':
-  //  Functions that are defined only for Double
+% if T in ['Double','CGFloat']:
+  //  Functions that are defined only for Double and CGFloat
```
This does run on 32-bit platforms, you know…

stephentyrone replied:
Yes. These interfaces are provided on 32b too.

jrose-apple replied:
Ah, the concern was that the results wouldn't match the tolerance.

stephentyrone replied:
No worries there; on 64b you get the Double implementation, so if it passes for Double it passes for CGFloat. On 32b, it's implemented by calling Double, then rounding to CGFloat, so if the Double implementation passed, the error of CGFloat is like < 0.50000003 ulp at *worst*.

stephentyrone committed:
Simply remove concrete functions replaced by identical generics...

... instead of marking them obsoleted, since the replacement is transparent to the user. Also fix a spelling error.

stephentyrone commented:
@swift-ci Please smoke test.
